core/IEnviroment.cs

```diff
@@ -0,0 +1,19 @@
+using System;
+
+namespace NosAyudamos
+{
+    public interface IEnviroment
+    {
+        string GetVariable(string name);
+    }
+
+    public class Enviroment : IEnviroment
+    {
+        public string GetVariable(string name)
+        {
+            Exceptions.ThrowIfNullOrEmpty(name, nameof(name));
+
+            return Environment.GetEnvironmentVariable(name);
```

Comment:
a more suscint way of doing this that doesn't require the "old-school" `Exceptions.ThrowIfNullOrEmpty` is two-fold: enable `<Nullable>Enable</Nullable>` in the project (therefore not needing to check for null), and then maybe an `Ensure.NotEmpty(name, nameof(name))` which returns de value or throws. 
With that in place, this implementation would just be:

```csharp
public string GetVariable(string name) => Environment.GetEnvironmentVariable(Ensure.NotEmpty(name, nameof(name)));
```